src/base/bittorrent/session.cpp

```diff
@@ -3068,6 +3070,8 @@ void Session::startUpTorrents()
         if (!addTorrent_impl(params.addTorrentData, params.magnetUri, TorrentInfo::loadFromFile(filePath), params.data))
             logger->addMessage(tr("Unable to resume torrent '%1'.", "e.g: Unable to resume torrent 'hash'.")
                                .arg(params.hash), Log::CRITICAL);
+        // don't waste memory with to many unprocessed queue entries
+        readAlerts();
```

Comment:
I think I was wrong at least partially. The GUI, and more specifically `TorrentModel::TorrentModel()` asks the session singleton for the torrents that it has and adds it to itself for GUI purposes.

So I think, at this point we are safe to handle alerts. I don't think there will be other alerts during startup that we care about.

But isn't it a little suboptimal to `readAlerts()` after every add? readAlerts() locks a mutex on our side, and probably libtorrent locks another when we tell it pop all alerts.
Maybe we should call readAlerts every 100 adds?